### PR TITLE
Feature/create hmrc api

### PR DESCRIPTION
**Pull request by mikeSmith08**
I still have to do the unit tests for the model objects, but I'll do that in another PR. The db side of things isn't fully complete either, but I'm unsure whether this should be done as a part of this story, or the story that starts checking against the db. I'll progress with it unless anyone thinks it should be in a different story.

### CODE REVIEW

**Comment by YetAnotherMatt** [comment]
The DB side of things will be done as part of HTBHF-638 - I've put some sql in the appropriate ticket based on the file format we got from HMRC.

File: api/src/main/java/uk/gov/dhsc/htbhf/hmrc/controller/HMRCEligibilityController.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+@RequestMapping("v1/hmrc/eligibility")
+@AllArgsConstructor
+@Slf4j
+@Api(description = "Endpoints for dealing with HMRC Eligibility requests.")
+public class HMRCEligibilityController {
+
+    private final EligibilityService eligibilityService;
+
+    @PostMapping
+    @ApiOperation("Retrieve the eligibility of a person for Child Tax Credits")
```
Comment (YetAnotherMatt) [inline api/src/main/java/uk/gov/dhsc/htbhf/hmrc/controller/HMRCEligibilityController.java]: This is actually retrieving their eligibility for Healthy Start, based on HMRC's opinion of their income and the fact that they're getting CTC

File: manifest.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  - java_buildpack
+  health-check-type: http
+  health-check-http-endpoint: /actuator/health
+  routes:
+    - route: htbhf-hmrc-api((app-suffix)).apps.internal
+  env:
+    JBP_CONFIG_OPEN_JDK_JRE: '{jre: { version: 11.+ }}'
+    APP_VERSION: ((app-version))
+  services:
+    - logit-ssl-drain
```
Comment (YetAnotherMatt) [inline manifest.yml:14]: We'll also need `    - htbhf-eligibility-api-postgres`

File: manifest/src/main/resources/manifest.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+---
```
Comment (YetAnotherMatt) [inline manifest/src/main/resources/manifest.yml]: This file should be ignored (manifest/src is in .gitignore but I guess you added this file before updating .gitignore)

File: api/src/test/java/uk/gov/dhsc/htbhf/hmrc/controller/HMRCEligibilityControllerIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    void shouldReturnBadRequestForInvalidEligibilityRequest() {
+        EligibilityRequest request = buildDefaultRequest().person(aPersonWithNoNino()).build();
+
+        ResponseEntity<ErrorResponse> errorResponse = restTemplate.postForEntity(ENDPOINT, request, ErrorResponse.class);
+
+        assertThat(errorResponse.getStatusCode()).isEqualTo(BAD_REQUEST);
+        assertValidationError(errorResponse, "person.nino", "must not be null");
+    }
+
+    private void assertValidationError(ResponseEntity<ErrorResponse> response, String field, String errorMessage) {
```
Comment (mikeSmith08) [inline api/src/test/java/uk/gov/dhsc/htbhf/hmrc/controller/HMRCEligibilityControllerIntegrationTest.java:63]: I've seen this method is a couple of projects now, shall I move out to common-test? My only worry is that this then couples `common-test` to `common-rest`, is that a problem? Thoughts?

Comment (YetAnotherMatt) [reply api/src/test/java/uk/gov/dhsc/htbhf/hmrc/controller/HMRCEligibilityControllerIntegrationTest.java:63]: common-rest and common-test are in the same repository so I don't think it should be an issue. If I understand correctly, common-rest would only have a test implementation dependency on common-test - which shouldn't have any impact on other components using common-rest.

File: manifest/src/main/resources/manifest.yml (hunk 3 above)
Comment (mikeSmith08) [reply manifest/src/main/resources/manifest.yml]: Yes, good spot, there were a few files that got committed before I sorted the .gitignore, I'll sort it.

File: api/src/test/java/uk/gov/dhsc/htbhf/hmrc/controller/HMRCEligibilityControllerIntegrationTest.java (hunk 4 above)
Comment (mikeSmith08) [reply api/src/test/java/uk/gov/dhsc/htbhf/hmrc/controller/HMRCEligibilityControllerIntegrationTest.java:63]: Yep, that was what I was thinking too but just wanted to get another opinion on it. I'll move it over then. :-)

**Review by gethyn1 — APPROVED** [review]

**Review by YetAnotherMatt — APPROVED** [review]

**Pull request merged**